src/db.ts

```diff
@@ -918,33 +917,40 @@ export abstract class Database {
           continue;
         }
 
-        updateJobs.recorded(values[1], values[0], values[4]);
-        batchedMs += 1;
-        batchedValues = batchedValues.concat(values);
-        if (batchedMs === Database.batchN) {
-          try {
-            const result = await this.recordMeasurementBatchedN(batchedValues);
-            recordedMeasurements += result;
-          } catch (err) {
-            // we may have concurrent inserts, or partially inserted data,
-            // where a request aborted
-            if (isUniqueViolationError(err)) {
-              recordedMeasurements += await this.recordMeasurementsFromBatch(
-                batchedValues
-              );
-            } else {
-              throw err;
-            }
+        let iMID = '';
+        // build a id for each unquie Measurement
+        iMID = iMID.concat(
+          run.id.toString() +
+            ' ' +
+            trial.id.toString() +
+            ' ' +
+            d.in.toString() +
+            ' ' +
+            criteria.get(m.c).id.toString()
```

Comment:
This feels like something where using format strings so `` `${run.id} ${trial.id} ${d.in}` ``.

Though, I don't really understand what the `iterationMap` is for.
What does it do? Why do you need construct these "ids" and then use this elaborate way to concatenate things?

Perhaps a comment and naming could help to clear things up?

Reply:
changed to 
`iMID = "${run.id} ${trial.id} ${d.in} ${criteria.get(m.c).id}";`
so what is iterationMap. before my change when we processed 50 datapoints from the payload we did an immediate batched insert at that point. 

problem with that is with the new ARRAY Value, i want to avoid updating the array for a given id to append on extra value. 
so once i proceed 50 datapoints i am not guaranteed that there is another value to be appended to the end of the array for a given ID we already seen ( an ID be the primary key for the measurements table which is {run.id} ${trial.id} ${d.in} ${criteria.get(m.c).id})

so what I do is map/dictionary of all ID’s and the array of values that goes with, this is (iterationMap).  Once there are no more data points to process I know I have all of the array values for all of the ID. So, then I can start batch inserting, safely knowing there will be no need to update an ID to append another value.

i will add a comment to make this clearer

Reply:
Hm, This method gets really large and hard to grasp.

Is this something you could refactor out into its own method, and do it as a conversion pass over the data before preparing the batched insertion?

This would mean two loops over the data, but that should be fine, and hopefully makes things more understandable.